crates/shared/src/zeroex_api.rs

```diff
@@ -118,6 +288,20 @@ pub trait ZeroExApi: Send + Sync {
     /// - https://0x.org/docs/guides/rfqt-in-the-0x-api
     /// - https://0x.org/docs/api#get-swapv1price
     async fn get_price(&self, query: SwapQuery) -> Result<PriceResponse, ZeroExResponseError>;
+
+    /// Retrieves specific page of current limit orders.
+    async fn get_orders_with_pagination(
+        &self,
+        query: &OrdersQuery,
+        results_per_page: usize,
+        page: usize,
+    ) -> Result<OrdersResponse, ZeroExResponseError>;
```

Comment:
Don't think this needs to be part of the `trait` if we only ever plan on calling the `get_orders` to get the full book.

Agree it makes 100% sense to have this method as an internal implementation detail though.